### PR TITLE
Bgp midonet

### PR DESCRIPTION
**Pull request by abelboldu**

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -31,5 +31,5 @@ midonet_gateway_veth1_ip: '{{ midonet_gateway_uplink_network | ipaddr("net") | i
 midonet_gateway_veth_prefix: '{{ midonet_gateway_uplink_network | ipaddr("host/prefix") | ipaddr("prefix") }}'
 midonet_gateway_bgp: False
 midonet_gateway_bgp_speaker_name: 'BGP-Speaker'
-midonet_gateway_bgp_speaker_local_as: 4000
+midonet_gateway_bgp_local_as: 4000
```
Comment (ryo-tagami) [inline defaults/main.yml]: It might be a good idea to use private AS number as a default
(MidoNet quick start guide uses `64512` as a local AS number).

**Review by ryo-tagami — APPROVED** [review]

File: defaults/main.yml (hunk 1 above)
Comment (abelboldu) [reply defaults/main.yml]: @ryo-tagami changing it to the internal one.

File: defaults/main.yml [hunk 2]
```diff
@@ -13,23 +13,23 @@ midonet_gateway_external_nic: '{{ ansible_default_ipv4.interface }}'
...
-midonet_gateway_external_subnet_gw: '{{ midonet_gateway_fip_net  | ipaddr("1") | ipaddr("address") }}'
+midonet_gateway_external_subnet_pool_start: '{{ midonet_gateway_external_network_cidr  | ipaddr("10") | ipaddr("address") }}'
+midonet_gateway_external_subnet_pool_end: '{{ midonet_gateway_external_network_cidr | ipaddr("254") | ipaddr("address") }}'
+midonet_gateway_external_subnet_gw: '{{ midonet_gateway_external_network_cidr  | ipaddr("1") | ipaddr("address") }}'
 midonet_gateway_edge_router_name: 'edge-router'
 midonet_gateway_edge_network_name: 'net-edge1-gw1'
 midonet_gateway_edge_subnet_name: 'subnet-edge1-gw1'
 midonet_gateway_edge_port_name: 'port-edge1-gw1'
-midonet_gateway_fip_net: '200.200.200.0/24'
+midonet_gateway_external_network_cidr: '200.200.200.0/24'
```
Comment (ryo-tagami) [inline defaults/main.yml:23]: Maybe changing this actually-allocated network (owned by Embratel, a Brazilian telco)
with a private network as well? The MidoNet quick start guide uses TEST-NET,
TEST-NET-2, TEST-NET-3 as an example:
> * One floating IP network
>   * 192.0.2.0/24
> * Two MidoNet gateway nodes
>   * gateway1, connecting to bgp1 via eth1
>   * gateway2, connecting to bgp2 via eth1
> * Two remote BGP peers
>   * bgp1, 198.51.100.1, AS 64513
>   * bgp2, 203.0.113.1, AS 64514
> * Corresponding MidoNet BGP peers
>   * 198.51.100.2, AS 64512
>   * 203.0.113.2, AS 64512

**Review by ryo-tagami — APPROVED** [review]

File: defaults/main.yml (hunk 2 above)
Comment (abelboldu) [reply defaults/main.yml:23]: Well, this is not meant to give outside connectivity at all, is an example.
The operations guide uses this CIDR, and it's in the puppet modules also, so I would like to keep it as is.

Comment (ryo-tagami) [reply defaults/main.yml:23]: OK, understood.

**Pull request merged**